### ISSUE TITLE
About BasicVSR! how can i train to get the same results for SR on REDS4

### ISSUE THREAD

**Issue by HappyXY** [open-mmlab/mmediting#447]
thanks for your code!
i try to train the basicvsr with reds4 on P100, i change the config file as follows subject to my GPU memory:  samples_per_gpu=2, num_input_frames=10
here is the loss curve and the final PSNR is about 30.3. any suggestion for training with small batchsize and frame sequence, thanks a lot!
 
![C7301B9D-8A80-4394-A47A-3AD01297B78B](https://user-images.githubusercontent.com/39117922/126733657-b307bac2-e6c1-4607-a401-52897493d053.png)

**Comment by ckkelvinchan** [comment]
You trained only for 60000 iterations, have you tried training for longer, say 300k? 

Another possible improvement is to find a training scheme more suitable for the reduced batch size and sequence length. But you may eventually get insignificant improvement, as it is normal that reducing the batch size and sequence length leads to a worse performance.

**Comment by HappyXY** [comment]
> You trained only for 60000 iterations, have you tried training for longer, say 300k?
> 
> Another possible improvement is to find a training scheme more suitable to the reduced batch size and sequence length. But you may eventually get insignificant improvement, as it is normal that reducing the batch size and sequence length leads to a worse performance.

thanks so much for your quick reply! 
i want to  consult you on another question, i try to use basicvsr to process the ' video compress restoration', but the training loss has no obviously decrease. i guess basivsr-net can solve some video restoration problem, including compression artifact reduction, but has no idea about this result. Hope to get some suggestion from you, thanks!

**Comment by ckkelvinchan** [comment]
> > You trained only for 60000 iterations, have you tried training for longer, say 300k?
> > Another possible improvement is to find a training scheme more suitable to the reduced batch size and sequence length. But you may eventually get insignificant improvement, as it is normal that reducing the batch size and sequence length leads to a worse performance.
> 
> thanks so much for your quick reply!
> i want to consult you on another question, i try to use basicvsr to process the ' video compress restoration', but the training loss has no obviously decrease. i guess basivsr-net can solve some video restoration problem, including compression artifact reduction, but has no idea about this result. Hope to get some suggestion from you, thanks!

I think BasicVSR can handle this task as well. In fact our latest model BasicVSR++ achieves very good results on this task. 

May I ask how you change the architecture? And is the visual quality improved?

**Comment by HappyXY** [comment]
> > > You trained only for 60000 iterations, have you tried training for longer, say 300k?
> > > Another possible improvement is to find a training scheme more suitable to the reduced batch size and sequence length. But you may eventually get insignificant improvement, as it is normal that reducing the batch size and sequence length leads to a worse performance.
> > 
> > 
> > thanks so much for your quick reply!
> > i want to consult you on another question, i try to use basicvsr to process the ' video compress restoration', but the training loss has no obviously decrease. i guess basivsr-net can solve some video restoration problem, including compression artifact reduction, but has no idea about this result. Hope to get some suggestion from you, thanks!
> 
> I think BasicVSR can handle this task as well. In fact our latest model BasicVSR++ achieves very good results on this task.
> 
> May I ask how you change the architecture? And is the visual quality improved?

Appreciate for your reply!
i just delete the upsample part as follows:
out = self.lrelu(self.fusion(out))
#out = self.lrelu(self.upsample1(out))
#out = self.lrelu(self.upsample2(out))
out = self.lrelu(self.conv_hr(out))
out = self.conv_last(out)
out += lr_curr
outputs[i] = out
the output result just be smoothed without any detail enhancement, and the psnr got a little increase
i also very interested in your work BasicVSR++ and the results of "NTIRE 2021 Challenge on Quality Enhancement of Compressed Video", and so i want to try the BasicVSR for this task, but did not get expected results.

**Comment by ckkelvinchan** [comment]
It seems there is no problem. There may be a chance that the optical flow estimation is more difficult for high-resolution images. In BasicVSR++, we first downsample the image 4 times. You may try this.

**Comment by HappyXY** [comment]
> It seems there is no problem. There may be a chance that the optical flow estimation is more difficult for high-resolution images. In BasicVSR++, we first downsample the image 4 times. You may try this.

thanks for your suggestion, i will try to add the downsample step.

**Comment by HappyXY** [comment]
> It seems there is no problem. There may be a chance that the optical flow estimation is more difficult for high-resolution images. In BasicVSR++, we first downsample the image 4 times. You may try this.

hi~ i try to use the pretrained net weight to test my compression data, but there are some artifact, and the artifact becomes bigger alone feature propogation, finally get a black image. i guess maybe the problem cause by the residual block without BN layer, do you got this error before and any suggestion to avoid this problem! 
Looking forward to your reply!
![00000056](https://user-images.githubusercontent.com/39117922/127947506-fc5198d7-3daf-4806-a2af-fd09bc118828.png)
 
![00000058](https://user-images.githubusercontent.com/39117922/127947588-1dbe0449-2b55-46e7-a93a-97fc411fc71c.png)

**Comment by ckkelvinchan** [comment]
This may be due to the mismatch of training and test distributions. And this kind of error could be amplified during propagation.

Usually, this can be resolved by
1. Training with longer intervals (as your test sequence is much longer)
2. Training with data that have the same distributions as the test data

I think normalization could also help, you can have a try.

**Comment by HappyXY** [comment]
> This may be due to the mismatch of training and test distributions. And this kind of error could be amplified during propagation.
> 
> Usually, this can be resolved by
> 
> 1. Training with longer intervals (as your test sequence is much longer)
> 2. Training with data that have the same distributions as the test data
> 
> I think normalization could also help, you can have a try.

thanks for your reply, i will try your suggestion!
